lib/src/pub_updater.dart

```diff
@@ -11,15 +11,24 @@ class PackageInfoRequestFailure implements Exception {}
 /// Exception thrown when the provided package information is not found.
 class PackageInfoNotFoundFailure implements Exception {}
 
+/// The pub.dev base url for querying package versions
+const _defaultBaseUrl = 'https://pub.dev/api/packages/';
+
 /// {@template pub_update}
 /// A Dart package which enables checking whether a package is up to date.
 /// {@endtemplate}
 class PubUpdater {
   /// {@macro pub_update}
-  PubUpdater([http.Client? client]) : _client = client;
-
-  /// The pub.dev base url for querying package versions
-  static const _baseUrl = 'https://pub.dev/api/packages/';
+  PubUpdater([http.Client? client, String customBaseUrl = _defaultBaseUrl])
+      : assert(
+          _isValidBaseURL(customBaseUrl),
+          '$customBaseUrl is not a valid pub api URL!',
```

Comment:
Instead of validating if it is valid base url, could we not just ask for the pub dev host and just suffix the `api/packages` whenever we call it?

Or even better, not validate it at all? As a proxy might be infront of it redirecting URLs that we know nothing about. Might be safer to assume that the user provided the correct URL and if they didn't, the request will thrown an error.

Reply:
Yeah I think we should configure this to be a `Uri` (instead of `String`) and remove the validation logic.